### PR TITLE
Fix #4168 Prevent login and edit user dialog hide on click out

### LINKED ISSUE #4168: Clicking on trasparent area close "Login" and "New User" modal windows
### Description
This could be a problem especially when the user, selecting the Usename or the Password, cliks on trasparent area (see image) 

Modal

### In case of Bug (otherwise remove this paragraph)

*Browser Affected*
(use this site: https://www.whatsmybrowser.org/ for non expert users)
- [X] Internet Explorer
- [X] Chrome
- [X] Firefox
- [X] Safari

*Browser Version Affected*

- Not browser related

*Steps to reproduce*

1.
- Open Mapstore
- Login

2.
- Open Mapstore
- Login
- Sing-in
- Manage Accounts
- New User

*Expected Result*

1. The modal window should not close with a click on trasparent area

2. The modal window should not close with a click on trasparent area

*Current Result*

1. The modal window closes with a click on trasparent area

2. The modal window closes with a click on trasparent area

### PR DESCRIPTION
**Pull request by kasongoyo**
## Description
see title

## Issues
 - #4168 
 - ...

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
see #4168 

**What is the new behavior?**
login and edit user modal does not hide on click out

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes, and I documented them in migration notes
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: web/client/components/manager/users/UserDialog.jsx [hunk 1]
```diff
@@ -223,7 +223,7 @@ class UserDialog extends React.Component {
     };
 
     render() {
-        return (<Dialog onClickOut={this.props.onClose} modal draggable={false} maskLoading={this.props.user && (this.props.user.status === "loading" || this.props.user.status === "saving")} id="mapstore-user-dialog" className="user-edit-dialog" style={assign({}, this.props.style, {display: this.props.show ? "block" : "none"})}>
```
Comment (MV88) [inline web/client/components/manager/users/UserDialog.jsx]: @kasongoyo can you provide motivation on why you think this should not be tested?

@offtherailz please follow this post

File: web/client/components/security/modals/LoginModal.jsx [hunk 2]
```diff
@@ -86,7 +86,7 @@ class LoginModal extends React.Component {
     };
 
     render() {
-        return (<Modal {...this.props.options} show={this.props.show} onHide={this.props.onClose}>
+        return (<Modal {...this.props.options} show={this.props.show} onHide={this.handleOnHide}>
```
Comment (MV88) [inline web/client/components/security/modals/LoginModal.jsx]: also GroupDialog should be included imho

File: web/client/components/security/modals/LoginModal.jsx [hunk 3]
```diff
@@ -99,6 +99,13 @@ class LoginModal extends React.Component {
         </Modal>);
     }
 
+    handleOnHide = (event) => {
```
Comment (MV88) [inline web/client/components/security/modals/LoginModal.jsx]: please add some more comments here
- onHide is called two times from Modal of bootstrap, one has the event one has not, specify this.

also onClose can share this method

File: web/client/components/manager/users/UserDialog.jsx (hunk 1 above)
Comment (kasongoyo) [reply web/client/components/manager/users/UserDialog.jsx]: @MV88 I think the hide/show behavior is full managed by Dialog component, here I was working on UserDialog build on top of Dialog and what I did was just removing the onClickOut prop. If we need to test hide/show behaviour, then we can do that on Dialog component, perhaps on a separate issue than this one.

Comment (offtherailz) [reply web/client/components/manager/users/UserDialog.jsx]: @kasongoyo I understand your point. 

The clickOut is already tested inside the Dialog component and I see you are removing the functionality, so in some way it's a feature removal, so testing that the feature is not there anymore is strange (as well as we will have also manual tests for them). 

I don't want to loose too much time in an issue like this (that for me is not even a bug :smile: ) 

The problem is that we want the bug fixes has at least some unit tests, that prevent in any case the other developers from introduce a regression. 

Long story short, if it too hard to provide a test for this, we can evaluate to skip. If it's easy, please add

Comment (kasongoyo) [reply web/client/components/manager/users/UserDialog.jsx]: @offtherailz thanks for detailed information. I have no problem adding tests until when it's hard to do so. The changes introduced are very less prone to regression because it's only affect single independent component so we can skip this.

**Comment by tdipisa** [comment]
@kasongoyo please, can you backport this to the stable branch 2019.02.xx?
